devel/fpm/Portfile

```diff
@@ -0,0 +1,55 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           compilers 1.0
+PortGroup           github 1.0
+
+github.setup        fortran-lang fpm 0.7.0 v
+revision            0
+categories          devel fortran
+license             MIT
+maintainers         {@barracuda156 gmail.com:vital.had} openmaintainer
+description         Fortran Package Manager
+long_description    Fortran Package Manager (fpm) is a package manager and build system for Fortran. \
+                    Its key goal is to improve the user experience of Fortran programmers. \
+                    It does so by making it easier to build your Fortran program or library, \
+                    run the executables, tests and examples, and distribute it as a dependency \
+                    to other Fortran projects.
+homepage            https://fpm.fortran-lang.org
+checksums           rmd160  3c0cc8372231118277f4bd6d280f14c576dfe6ab \
+                    sha256  f8d6c469bfdbecb1fbed5c5f182dcd76bc052fe6a1c485e3e7e076515ab48f52 \
+                    size    169451
+
+depends_build-append \
+                    port:curl
+
+# Change this, once older OSs are moved to gcc12/13.
+platform darwin {
+    if {${os.major} < 10} {
+        depends_run-append  port:gcc7
+    } else {
+        depends_run-append  port:gcc12
+    }
+}
+
+patchfiles          patch-install.diff \
+                    patch-filesystem_utilities.diff
+
+post-patch {
+    reinplace "s,@FC@,${configure.fc}," ${worksrcpath}/install.sh
+    reinplace "s,@PREFIX@,${destroot}${prefix}," ${worksrcpath}/install.sh
+    reinplace "s,@VERSION@,${version},g" ${worksrcpath}/install.sh
+}
+
+compiler.blacklist-append \
+                    *gcc-4.* macports-gcc-5 macports-gcc-6
+compilers.setup     require_fortran
+
+use_configure       no
+
+build.cmd           ${worksrcpath}/install.sh
+build.target
+build.cmd-prepend   FPM_CFLAGS="${configure.cflags} [get_canonical_archflags cc]" \
+                    FPM_LDFLAGS="${configure.ldflags} [get_canonical_archflags ld]"
+
+destroot            {}
```

Comment:
So it looks like this Portfile is installing straight into the destroot path during the `build` phase, and so doesn't need to do anything during the `destroot` phase.  Is there any way that this can be changed?

Does `install.sh` only copy files? If that's the case, then it will make more sense to have an empty 'build' phase and copy files in the `destroot` block.

Reply:
The thing is that `$BOOTSTRAP_DIR/fpm install --compiler "$FC" --flag "$FFLAGS" --prefix "$PREFIX"` both builds a full version of `fpm` and installs it into specified prefix. Perhaps this can be moved into destroot stage, but that gonna be a part of building. In either case stages are not neat – but that is the case with quite a number of ports, in fact.

Let me try modifying the script to see if doing `fpm install` in destroot works. Otherwise perhaps prefix can be redefined and destroot implemented manually.

Reply:
Well if it's actually building the software, then it should be left as-is, in the `build` phase.

The problem is that things are being installed into the `destroot` before the `destroot` phase gets run, where accessory files are installed before the software is to be staged into the `destroot`.  I guess if you see this in other ports, we can leave it, but it's not optimal.

Reply:
@herbygillot Once CI pass, should be good, I think.

Reply:
@herbygillot Okay, now hopefully everything works as supposed. Took a bit to figure out how to make it correctly in the PG.